### PR TITLE
[bitnami/magento] Add strategy type to Magento deployment

### LINKED ISSUE #2435: [Magento] allow setting 'StrategyType'
**Which chart**:
Magento

**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I would like to set the strategy type for my chart to 'Recreate'. 

**Additional context**
I am using Azure Disk for persistent storage and using 'RollingUpdate' sometimes keeps the pod in pending state because it schedules the new pod on a new VM which fails on mounting the disk due that it's already mounted to another VM.

### PR DESCRIPTION
**Pull request by stijnbrouwers**
**Description of the change**

Adds 'strategy.type' to the magento deployment

**Benefits**

Except from using the default 'RollingUpdate' it allows using 'Recreate'

**Applicable issues**

  - fixes #2435 

**Checklist**
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/magento/README.md [hunk 1]
```diff
@@ -137,6 +137,7 @@ The following table lists the configurable parameters of the Magento chart and t
 | `persistence.magento.storageClass`      | PVC Storage Class for Magento volume                                                                 | `nil`  (uses alpha storage annotation)                       |
 | `persistence.magento.accessMode`        | PVC Access Mode for Magento volume                                                                   | `ReadWriteOnce`                                              |
 | `persistence.magento.size`              | PVC Storage Request for Magento volume                                                               | `8Gi`                                                        |
+| `strategy.type`                         | Deployment's strategy type                                                                           | `RollingUpdate`                                              |
```
Comment (carrodher) [inline bitnami/magento/README.md:140]:
```suggestion
| `updateStrategy`                         | Set to Recreate if you use persistent volume that cannot be mounted by more than one pods                                                                 | `RollingUpdate`                                              |
```

File: bitnami/magento/values-production.yaml [hunk 2]
```diff
@@ -265,6 +265,11 @@ persistence:
     accessMode: ReadWriteOnce
     size: 8Gi
 
+## Configure strategy type
+## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+strategy:
+  type: RollingUpdate
```
Comment (carrodher) [inline bitnami/magento/values-production.yaml:271]:
```suggestion
 ## Set to Recreate if you use persistent volume that cannot be mounted by more than one pods
 ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
 updateStrategy: RollingUpdate
```

File: bitnami/magento/values.yaml [hunk 3]
```diff
@@ -265,6 +265,11 @@ persistence:
     accessMode: ReadWriteOnce
     size: 8Gi
 
+## Configure strategy type
+## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+strategy:
+  type: RollingUpdate
```
Comment (carrodher) [inline bitnami/magento/values.yaml:271]:
```suggestion
 ## Set to Recreate if you use persistent volume that cannot be mounted by more than one pods
 ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
 updateStrategy: RollingUpdate
```

File: bitnami/magento/templates/deployment.yaml [hunk 4]
```diff
@@ -13,6 +13,8 @@ spec:
     matchLabels:
       app: {{ template "magento.fullname" . }}
       release: "{{ .Release.Name }}"
+  strategy:
+    type: "{{ .Values.strategy.type }}"
```
Comment (carrodher) [inline bitnami/magento/templates/deployment.yaml:17]:
```suggestion
    type: {{ .Values.updateStrategy }}
```

**Comment by carrodher** [comment]
Thanks for your contribution 👏

File: bitnami/magento/README.md (hunk 1 above)
Comment (javsalgar) [reply bitnami/magento/README.md:140]: +1

**Comment by stijnbrouwers** [comment]
@carrodher , thanks for the review. Looks perfect to go!

**Comment by carrodher** [comment]
I am launching the PR in our CI/CD pipeline if everything works as expected the new version of the chart should be released soon

**Pull request merged**